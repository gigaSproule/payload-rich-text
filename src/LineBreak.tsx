import type { Options } from "./types";

export interface LineBreakData {
  type: "linebreak";
  version: number;
}

export type Props = {
  data: LineBreakData;
  options?: Options;
};

export const LineBreak = ({ data, options }: Props) =>
  options?.lineBreak ? options.lineBreak(data) : <br />;
