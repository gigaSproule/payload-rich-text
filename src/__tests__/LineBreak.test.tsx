import { describe, expect, it } from "@jest/globals";
import { render } from "@testing-library/react";
import type { Props } from "../LineBreak";
import { LineBreak } from "../LineBreak";
import { createLineBreakData } from "./helpers/createLineBreakData";

describe("LineBreak", () => {
  it("should render default br if no options provided", () => {
    const lineBreakData: Props["data"] = createLineBreakData();
    const { container } = render(<LineBreak data={lineBreakData} />);
    expect(container).toMatchSnapshot();
  });

  it("should render default br if options provided without lineBreak", () => {
    const lineBreakData: Props["data"] = createLineBreakData();
    const { container } = render(
      <LineBreak data={lineBreakData} options={{}} />,
    );
    expect(container).toMatchSnapshot();
  });

  it("should render custom render if options provided with lineBreak", () => {
    const lineBreakData: Props["data"] = createLineBreakData();
    const { container } = render(
      <LineBreak
        data={lineBreakData}
        options={{
          lineBreak: (lineBreak) => {
            return <div>Custom BR - {JSON.stringify(lineBreak)}</div>;
          },
        }}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
