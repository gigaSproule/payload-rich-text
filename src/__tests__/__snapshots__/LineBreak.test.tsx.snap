// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineBreak should render custom render if options provided with lineBreak 1`] = `
<div>
  <div>
    Custom BR - 
    {"type":"linebreak","version":1}
  </div>
</div>
`;

exports[`LineBreak should render default br if no options provided 1`] = `
<div>
  <br />
</div>
`;

exports[`LineBreak should render default br if options provided without lineBreak 1`] = `
<div>
  <br />
</div>
`;
